evaluation-tasks/util/s3.py

```diff
@@ -0,0 +1,153 @@
+"""
+Utility functions for interacting with an S3 bucket.
+"""
+
+import os
+import subprocess
+
+import boto3
+from luigi.contrib.s3 import S3Client
+from botocore.client import ClientError
+import luigi
+
+from .luigi import WorkTask
+
+
+class CacheTarCorpus(WorkTask):
+    """
+    If the tar file is cached in S3, we simply retrieve it and untar
+    it, and skip the pipeline.
+
+    If the tar file is NOT in S3, we run the pipeline, create the
+    tar-file, and upload it to S3.
+    """
+
+    task_name = luigi.Parameter()
+    bucket = luigi.Parameter()
+    region = luigi.Parameter()
+    next_task = luigi.Parameter()
+    # TODO: How do we support params for next_task??
```

Comment:
I'm passing in the final task uninitialized -- so maybe it just needs to be initialized first -- or a list of next tasks that are all initialized instead of futzing with initializing them with params if they need them.

Reply:
@jorshi I'm sort of leaning towards us just having completely separate scripts from S3 sync. Thoughts?

And not wrapping it up with the luigi logic, which is already relatively complicated

Reply:
This pattern of passing a task in as a Luigi parameter, I haven't used before

Reply:
Agreed. Will pull it out and make an issue for creating a separate script for s3 sync

Reply:
https://github.com/neuralaudio/hear2021-eval-kit/issues/11